src/aspire/source/simulation.py

```diff
@@ -125,10 +127,11 @@ def __init__(
         self.offsets = offsets
         self.amplitudes = amplitudes
 
-        self.noise_adder = None
-        if noise_filter is not None and not isinstance(noise_filter, ZeroFilter):
-            logger.info("Appending a NoiseAdder to generation pipeline")
-            self.noise_adder = NoiseAdder(seed=self.seed, noise_filter=noise_filter)
+        if noise_adder is not None:
+            logger.info(f"Appending {noise_adder} to generation pipeline")
```

Comment:
We should give `NoiseAdder` a more descriptive `repr`. At least what sort of filter it was instantiated with?

Reply:
Okay. Made a repr and a str method. (str will get used here).

Printing the filter is likely to be useless, but I'll add it now in case that changes...